src/core/ext/transport/chttp2/client/chttp2_connector.cc

```diff
@@ -205,7 +211,15 @@ void Chttp2Connector::OnReceiveSettings(void* arg, grpc_error_handle error) {
         self->result_->Reset();
       }
       self->MaybeNotify(error);
-      grpc_timer_cancel(&self->timer_);
+      if (self->timer_handle_.has_value()) {
+        if (self->args_.channel_args.GetObject<EventEngine>()->Cancel(
```

Comment:
We probably don't want to be digging this out of channel args repeatedly

Reply:
Added an `event_engine_` member, PTAL.